site/content/docs/development/architecture.md

```diff
@@ -0,0 +1,142 @@
+| ----------- | ----------- |
+| Config | [ClusterConfigTemplate](reference/#clusterconfigtemplate) |
+| Image | [ClusterImageTemplate](reference/#clusterimagetemplate) |
+| Source | [ClusterSourceTemplate](reference/#clustersourcetemplate) |
+| Deployment | [ClusterDeploymentTemplate](reference/#clusterdeploymenttemplate) |
+| | [ClusterTemplate](reference/#clustertemplate) |
+
+### Owners
+
+Owners represent the workload or deliverable, which in many cases refer to a single application's source or image 
+location.
+
+Owners are the developer provided configuration which cause a blueprint to be reconciled into resources.
+
+They consist of:
+* **Labels**: blueprints will select based on the labels of an owner, see [selectors](#selectors) 
+* **Params**: parameters supplied to the blueprint, see [Parameter Hierarchy](#parameter-hierarchy)
+* **Source**: The source reference for the input to the Supply Chain or Delivery Blueprints,
+see [Workload](reference.md/#workload) and [Deliverable](reference.md/#deliverable)
+
+![Owner](../img/owner.jpg)
+
+| Owner      | Blueprint |
+| ----------- | ----------- |
+| Workload | ClusterSupplyChain |
+| Deliverable | ClusterDelivery |
+
+## Theory of operation
+
+Given an owner that matches a blueprint, cartographer reconciles the resources referenced by the blueprint.
+The resources are only created when the inputs are satisfied, and a resource is only updated when it's inputs change.
+This results in a system where a new intrinsic result from one resource can cause other resources to change.
+
+Although Cartographer is not a 'runner of things', a resource can be something as simple as a Job or a CI pipeline.
+However, one advantage of Cartographer's design, is that a resource can also be untriggered. Imagine a Build resource 
+that discovers new base OCI images. If it rebuilds your image, then cartographer will see this new image and update 
+other linked resources.
+
+![Generic Blueprint](../img/generic.jpg)
+<!-- https://miro.com/app/board/uXjVOeb8u5o=/ -->
+
+When Cartographer reconciles an owner, each resource in the matching blueprint is reconciled:
+
+1. **Generate Inputs**: Using the **blueprint resource's** `inputs` as a reference, select outputs from previously applied **Kubernetes resources**
+2. **Generate Params**: Using the [Parameter Hierarchy](architecture.md#parameter-hierarchy), generate parameter values   
+3. **Generate and apply resource spec**: Apply the result of interpolating `spec.template` (or `spec.ytt`), **inputs**, **params** and the **owner spec**. 
+4. **Retrieve Output**: Store the output from the applied resource. The output to use is specified in the **template output path**.  
+
+![Realize](../img/realize.jpg)
+
+### Complete Supply Chain and Delivery with GitOps
+
+![Gitops](../img/gitops.jpg)
+
+## Blueprint Details
+
+### ClusterSupplyChain
+A ClusterSupplyChain blueprint continuously integrates and builds your app.
+
+![ClusterSupplyChain](../img/supplychain.jpg)
```

Comment:
suggestion:
This diagram is a bit blurry and the writing is quite small so it is tough to read. I can tell there is a lot to get in there, but is there a way to make it a bit more readable?

Reply:
In time we will move these to svg and rework them.

Reply:
A picture is worth a thousand words (and this is a good picture). This could still benefit from some text. Particularly to call out that the handoff is always by configuring the supply chain to put configuration in some place that the Deliverable can read it (On a single cluster, that could be a ConfigMap). And to call out that in the multi-cluster case we expect users to use GitOps with either a git repo or an image registry.